### PR TITLE
Shiny volume support

### PR DESCRIPTION
**Pull request by alfredeen**
## Description

This PR adds the ability to mount a volume to Shiny proxy apps. Currently this is a feature for admins. The UI does not contain input field for path nor volume.

Additionally:
- Changed Foreign key delete logic for the dash, shiny and custom app types. Now if the volume is deleted app will not be deleted.
- Changed how volumes are displayed in the admin panel. Added to what project it belongs to as now admins can attach volumes to apps from any other project. Now it looks like this:
image

Jira [link](https://scilifelab.atlassian.net/browse/SS-1061)

## Checklist

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [x] I have included a link to the issue on GitHub or JIRA (if any)
- [x] I have included migration files (if there are changes to the model classes)
- [ ] I have included, reviewed and executed tests (unit and end2end) to complement my changes
- [ ] I have updated the related documentation (if necessary)
- [x] I have added a reviewer for this pull request
- [x] I have added myself as an author for this pull request
- [ ] In the case I have modified settings.py, then I have also updated the studio-settings-configmap.yaml file in serve-charts

### CODE REVIEW

File: apps/models/app_types/shiny.py [hunk 1]
```diff
@@ -23,16 +23,21 @@ class ShinyInstance(BaseAppInstance, SocialMixin, LogsEnabledMixin):
         ("public", "Public"),
         ("link", "Link"),
     )
+
+    volume = models.ForeignKey(
+        "VolumeInstance", blank=True, null=True, related_name="%(class)s", on_delete=models.CASCADE
```
Comment (churnikov) [inline apps/models/app_types/shiny.py:28]: Just want to make sure, that this is the behavior that we want.

By setting `on_delete=models.CASCADE` when volume is deleted, this app will be deleted as well. It seems like we might not want that based on the recent changes with the subdomains

image

[PR](https://github.com/ScilifelabDataCentre/stackn/pull/211/files#diff-1f62b2845f7575d4e30f0a07f7cb8f881419f886f178fdefa619a1f5293b24a7)

File: apps/models/app_types/shiny.py [hunk 2]
```diff
@@ -23,16 +23,21 @@ class ShinyInstance(BaseAppInstance, SocialMixin, LogsEnabledMixin):
...
     proxy = models.BooleanField(default=True)
     container_waittime = models.IntegerField(default=20000)
     heartbeat_timeout = models.IntegerField(default=60000)
     heartbeat_rate = models.IntegerField(default=10000)
-    minimum_seats_available = models.IntegerField(default=2)
-    seats_per_container = models.IntegerField(default=3)
-    allow_container_reuse = models.BooleanField(default=True)
+    minimum_seats_available = models.IntegerField(default=1)
+    seats_per_container = models.IntegerField(default=1)
+    allow_container_reuse = models.BooleanField(default=False)
```
Comment (churnikov) [inline apps/models/app_types/shiny.py:40]: Information on this is possible to find via PRs and jira, but for convenience, maybe we'll add a comment with a short description what do these settings do?

And I think that's just me not knowing how the whole apps deployment pipeline work, but how does it play with the [values file in the shiny proxy chart](https://github.com/ScilifelabDataCentre/serve-charts/pull/49/files#diff-11cbabbb8a92fa9b735ed4f34d7ae2575c2ab1f62b5d6659f323eb069d530f8c)?

File: apps/models/app_types/shiny.py (hunk 1 above)
Comment (alfredeen) [reply apps/models/app_types/shiny.py:28]: True. I will change this. I think custom apps have the same cascade delete.

File: apps/models/app_types/shiny.py (hunk 2 above)
Comment (alfredeen) [reply apps/models/app_types/shiny.py:40]: Good point. I will add some clarifying comments. These values override the chart default values.

File: apps/models/app_types/custom.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ class CustomAppInstance(BaseAppInstance, SocialMixin, LogsEnabledMixin):
     )
 
     volume = models.ForeignKey(
-        "VolumeInstance", blank=True, null=True, related_name="%(class)s", on_delete=models.CASCADE
+        "VolumeInstance", blank=True, null=True, related_name="%(class)s", on_delete=models.SET_NULL
```
Comment (churnikov) [inline apps/models/app_types/custom.py:28]: Shouldn't this have generated migrations files?

Comment (alfredeen) [reply apps/models/app_types/custom.py:28]: Yes indeed. Coming...

**Review by churnikov — APPROVED** [review]

**Comment by churnikov** [comment]
I'll be honest, I haven't tried it locally, but if it works for you and tests are passing then LGTM

**Review by churnikov — APPROVED** [review]
Looked through it again. 
I think it looks good

**Pull request merged**